From 4bef868317872c79a4e1910f0bbe65b6654307c1 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Thu, 5 Feb 2026 00:58:49 +0100
Subject: [PATCH] Prefer Wayland over X11

Not specifying the SDL_VIDEODRIVER hint will cause SDL to only use X11
and not even try Wayland, which is a problem on systems that only have
Wayland.

By specifying SDL_VIDEODRIVER="windows,wayland,x11" the game will work
on both Wayland and X11 sessions as well as on Windows.

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 src/main.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index e54ff7e88..76bd89a39 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -32,6 +32,10 @@ extern "C" int main(int argc, char **argv)
 	Profiler::detect(argc, argv);
 #endif
 
+	/* SDL2 does not enable Wayland by default, but we want to prefer it
+	 * when available, over X11 */
+	SDL_setenv("SDL_VIDEODRIVER", "windows,wayland,x11", 0);
+
 	OS::SetDPIAware();
 
 	RunMode mode = MODE_GAME;
-- 
2.51.0

